Failed to include <thr/xthreads.h> on Windows

<!--- Provide a general summary of the issue in the Title above -->
Precisely what the title says. Any attempt to build fails as a result.

## Expected Behavior
<!--- Tell us what should happen -->
Builds should not fail.

## Current Behavior
<!--- Tell us what happens instead of the expected behavior -->
Builds fail with lots of:

```console
../utils/TimedMutex.hpp(26,10): fatal error C1083: Cannot open include file: 'thr/xthreads.h': No such file or directory
```

## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. Follow [ROS 2 source install instructions for Windows](https://index.ros.org/doc/ros2/Installation/Eloquent/Windows-Development-Setup/).
2. Run `colcon build --packages-up-to fastrtps`

## System information
<!--- Provide information about your system environment. -->
- Fast-RTPS version: `v1.9.x`  (though I'd expect it to fail on `master` too) <!--- e.g. commit hash, tag, branch, ... -->
- OS: Windows Server 2019
- ROS2: Eloquent

## Additional context
<!--- Add any context about the problem here. -->
To build, I'm using latest Visual Studio 2019 (MSVC 14.24.28314) and CMake 3.16.1. Digging a bit, in this particular MSVC version the `thr/` subdirectory is gone. Changing `<thr/xthreads.h>` to `<xthreads.h>` fixes the problem.

Interestingly enough, there's not much information online about this particular header file. It'd almost seem it's an internal API header, which would explain why it changed without notice.

## Additional resources

--- a/include/fastrtps/utils/TimedMutex.hpp
+++ b/include/fastrtps/utils/TimedMutex.hpp
@@ -23,8 +23,10 @@
 #include <iostream>
 
 #if defined(_WIN32)
-#include <thr/xthreads.h>
-extern int clock_gettime(int, struct timespec* tv);
+#include <thread>
+extern int clock_gettime(
+        int,
+        struct timespec* tv);
 #elif _GTHREAD_USE_MUTEX_TIMEDLOCK
 #include <mutex>
 #else
@@ -44,8 +46,10 @@ public:
         _Mtx_init(&mutex_, _Mtx_timed);
     }
 
-    TimedMutex(const TimedMutex&) = delete;
-    TimedMutex& operator=(const TimedMutex&) = delete;
+    TimedMutex(
+            const TimedMutex&) = delete;
+    TimedMutex& operator =(
+            const TimedMutex&) = delete;
 
     ~TimedMutex()
     {
@@ -63,14 +67,15 @@ public:
     }
 
     template <class Rep, class Period>
-    bool try_lock_for(const std::chrono::duration<Rep, Period>& rel_time)
+    bool try_lock_for(
+            const std::chrono::duration<Rep, Period>& rel_time)
     {
         return try_lock_until(chrono::steady_clock::now() + rel_time);
     }
 
     template <class Clock, class Duration>
     bool try_lock_until(
-        const std::chrono::time_point<Clock, Duration>& abs_time)
+            const std::chrono::time_point<Clock, Duration>& abs_time)
     {
         std::chrono::nanoseconds nsecs = abs_time - std::chrono::steady_clock::now();
 
@@ -110,8 +115,10 @@ public:
         _Mtx_init(&mutex_, _Mtx_timed | _Mtx_recursive);
     }
 
-    RecursiveTimedMutex(const TimedMutex&) = delete;
-    RecursiveTimedMutex& operator=(const TimedMutex&) = delete;
+    RecursiveTimedMutex(
+            const TimedMutex&) = delete;
+    RecursiveTimedMutex& operator =(
+            const TimedMutex&) = delete;
 
     ~RecursiveTimedMutex()
     {
@@ -129,14 +136,15 @@ public:
     }
 
     template <class Rep, class Period>
-    bool try_lock_for(const std::chrono::duration<Rep, Period>& rel_time)
+    bool try_lock_for(
+            const std::chrono::duration<Rep, Period>& rel_time)
     {
         return try_lock_until(chrono::steady_clock::now() + rel_time);
     }
 
     template <class Clock, class Duration>
     bool try_lock_until(
-        const std::chrono::time_point<Clock, Duration>& abs_time)
+            const std::chrono::time_point<Clock, Duration>& abs_time)
     {
         std::chrono::nanoseconds nsecs = abs_time - std::chrono::steady_clock::now();
         if (0 < nsecs.count())
@@ -148,7 +156,7 @@ public:
             nsecs -= secs;
             max_wait.tv_sec += secs.count();
             max_wait.tv_nsec = (long)nsecs.count();
-            return (_Thrd_success == _Mtx_timedlock(mutex_, (xtime*)& max_wait));
+            return (_Thrd_success == _Mtx_timedlock(mutex_, (xtime*)&max_wait));
         }
         else
         {
@@ -178,8 +186,10 @@ public:
         pthread_mutex_init(&mutex_, nullptr);
     }
 
-    TimedMutex(const TimedMutex&) = delete;
-    TimedMutex& operator=(const TimedMutex&) = delete;
+    TimedMutex(
+            const TimedMutex&) = delete;
+    TimedMutex& operator =(
+            const TimedMutex&) = delete;
 
     ~TimedMutex()
     {
@@ -197,14 +207,15 @@ public:
     }
 
     template <class Rep, class Period>
-    bool try_lock_for(const std::chrono::duration<Rep, Period>& rel_time)
+    bool try_lock_for(
+            const std::chrono::duration<Rep, Period>& rel_time)
     {
         return try_lock_until(std::chrono::steady_clock::now() + rel_time);
     }
 
     template <class Clock, class Duration>
     bool try_lock_until(
-        const std::chrono::time_point<Clock, Duration>& abs_time)
+            const std::chrono::time_point<Clock, Duration>& abs_time)
     {
         std::chrono::nanoseconds nsecs = abs_time - std::chrono::steady_clock::now();
         struct timespec max_wait = { 0, 0 };
@@ -238,8 +249,10 @@ public:
         pthread_mutex_init(&mutex_, &mutex_attr_);
     }
 
-    RecursiveTimedMutex(const RecursiveTimedMutex&) = delete;
-    RecursiveTimedMutex& operator=(const RecursiveTimedMutex&) = delete;
+    RecursiveTimedMutex(
+            const RecursiveTimedMutex&) = delete;
+    RecursiveTimedMutex& operator =(
+            const RecursiveTimedMutex&) = delete;
 
     ~RecursiveTimedMutex()
     {
@@ -258,14 +271,15 @@ public:
     }
 
     template <class Rep, class Period>
-    bool try_lock_for(const std::chrono::duration<Rep, Period>& rel_time)
+    bool try_lock_for(
+            const std::chrono::duration<Rep, Period>& rel_time)
     {
         return try_lock_until(std::chrono::steady_clock::now() + rel_time);
     }
 
     template <class Clock, class Duration>
     bool try_lock_until(
-        const std::chrono::time_point<Clock, Duration>& abs_time)
+            const std::chrono::time_point<Clock, Duration>& abs_time)
     {
         std::chrono::nanoseconds nsecs = abs_time - std::chrono::steady_clock::now();
         struct timespec max_wait = { 0, 0 };

